src/pathfora.js

```diff
@@ -2875,7 +2875,19 @@
 
       this.getData(recommendUrl, function (json) {
         var resp = JSON.parse(json);
```

Comment:
considering this is our api it shouldn't be an issue but might be worth doing a try/catch here to handle the edge case of invalid json. at the moment if that api returns invalid json it will throw an error. seems like in that case we'd want to raise a warning and then return `[]`